### ISSUE TITLE
Please clarify install instructions and requirements (docs say TBD a lot)

### ISSUE THREAD

**Issue by bix6** [ansible-collections/community.libvirt#71]
##### SUMMARY

Please clarify install instructions and requirements. I am struggling to setup this module and cannot find clear documentation anywhere. I am attempting to hack multiple guides together and still not having success. I have tried both python2 and python3 now.

Here is the current flow I am attempting. It's still not working but I don't know where to go from here.

```
sudo apt update
sudo apt -y install python3-pip
python3 -m pip install -U pip
python3 -m pip install --user ansible
// I think pkg-config may come with libvirt-dev as well
sudo apt install pkg-config
sudo apt install libvirt-dev
python3 -m pip install libvirt-python

// Update path
nano ~/.bashrc
// Add this
export PATH=$PATH:$HOME/.local/bin
// Refresh bash
source ~/.bashrc
```

This is the error I have now in my playbook (I have read every page I can find discussing this error):
```
{"changed": false, "msg": "The `libvirt` module is not importable. Check the requirements."}
```

##### ISSUE TYPE
- Documentation Report

##### COMPONENT NAME

community.libvirt.virt

##### ANSIBLE VERSION

```paste below
[DEPRECATION WARNING]: Ansible will require Python 3.8 or newer on the controller starting with Ansible 2.12. Current 
version: 3.6.9 (default, Jan 26 2021, 15:33:00) [GCC 8.4.0]. This feature will be removed from ansible-core in version 
2.12. Deprecation warnings can be disabled by setting deprecation_warnings=False in ansible.cfg.
ansible [core 2.11.1] 
  config file = None
  configured module search path = ['/home/ubuntu/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/ubuntu/.local/lib/python3.6/site-packages/ansible
  ansible collection location = /home/ubuntu/.ansible/collections:/usr/share/ansible/collections
  executable location = /home/ubuntu/.local/bin/ansible
  python version = 3.6.9 (default, Jan 26 2021, 15:33:00) [GCC 8.4.0]
  jinja version = 2.10
  libyaml = True
```

**Comment by ceesios** [comment]
Hi,
I ran into the same problem today. I overlooked the fact that ansible is running on the remote host, and that is where the requirements should be installed. So either add a task to install the requirements remotely or use delegate_to to run the task locally.

**Comment by dtantsur** [comment]
On top of that, could you turn "Check the requirements" into something more verbose? Maybe the actual ImportError?

We have a very recent breakage on CentOS 8 (Stream), and I don't know what causes it. The requirements used to be fine just a couple of weeks ago.

**Comment by Andersson007** [comment]
@dtantsur hi, thanks for reporting this!
You installed libvirt-python via pip3. Could you please try run your playbook with the ` -e "ansible_python_interpreter=/path/to/your/python3/interpreter"` argument?
Looks like the module uses python2 and doesn't see python3 libs.

I'm looking forward to your feedback

**Comment by dtantsur** [comment]
@Andersson007 that's what we're doing. We're well past Python 3 issues at this point (we switched more than a year ago).

This is a workaround we have to provide: https://review.opendev.org/c/openstack/bifrost/+/800214/3/playbooks/roles/bifrost-create-vm-nodes/tasks/prepare_libvirt.yml. It essentially prevents `pip` from using `libvirt` wheels and forces a source installation. I must admit, I have no clue how and why it works, especially since this issue is not trivial to reproduce (we hit it in the CI workers, but not locally).

**Comment by Andersson007** [comment]
@dtantsur thank you for the feedback and the workaround!
I pinged the folks in https://github.com/ansible-collections/community.libvirt/pull/77#issuecomment-882513098. Maybe they have some ideas and will respond. We should definitely improve this part of the doc.

**Comment by odyssey4me** [comment]
I would caution everyone that mixing your system packages and packages from pypi is not a good idea. Things from pypi should be installed in a user's home directory (not root) using --user or should go into a venv.

libvirt-python is a bit of a pain because it has host OS binary bindings, so usually it is best to install it via the distribution packaging (a package called python-libvirt or something like that). This package must be on the *target* host where the virtual machines will be run or where the inventory source is.

**Comment by Andersson007** [comment]
Could anyone try the way suggested by @odyssey4me and give feedback?
If we had a step-by-step guide, we could create INSTALLATION file and link to it from README (or put the guide directly to README if it's not long).

**Comment by dtantsur** [comment]
> I would caution everyone that mixing your system packages and packages from pypi is not a good idea. Things from pypi should be installed in a user's home directory (not root) using --user or should go into a venv.

Yep, except that python-apt/python-dnf are not distributed via pypi, so to use ansible with venv you need to use `--system-site-packages`. Which could contribute to our issues :(

**Comment by Andersson007** [comment]
(Off topic: there's the [Quick-start guide](https://github.com/ansible/community-docs/blob/main/create_pr_quick_start_guide.rst) there which can help people who want to clarify the installation process but have no experience in contributing to Ansible and how to submit a pull request. All related to testing in the guide is irrelevant here)

**Comment by bix6** [comment]
Just dropping in a comment since I opened this issue and I won’t be checking back on this as I’ve found manual and shell script workarounds. Glad I raised this comment though! Sounds like my issue could be needing some package installed on the target server as I only installed on Ansible host.
